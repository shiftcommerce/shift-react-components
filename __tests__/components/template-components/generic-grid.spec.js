// Libraries
import React from 'react'

// Components
import GenericGrid from '../../../src/components/template-components/generic-grid'

// Fixtures
import genericGridData from '../../fixtures/generic-grid'

test('GenericGrid component renders correctly', () => {
  // Arrange & Act
  const wrapper = mount(
    <GenericGrid
      componentData={genericGridData}
    />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(genericGridData.header)
  expect(wrapper).toIncludeText(genericGridData.slide1_text)
  expect(wrapper).toIncludeText(genericGridData.slide2_text)
  expect(wrapper).toIncludeText(genericGridData.slide3_text)

  expect(wrapper.find('a').first()).toHaveProp('href', genericGridData.slide1_url[0].canonical_path)
  expect(wrapper.find('.o-template-component__cat-button').children()).toIncludeText(genericGridData.cat_text)
})
