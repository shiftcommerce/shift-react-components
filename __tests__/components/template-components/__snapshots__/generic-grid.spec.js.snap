// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericGrid component renders correctly 1`] = `
<GenericGrid
  componentData={
    Object {
      "cat_text": "Cat Button",
      "cat_url": Array [
        Object {
          "canonical_path": "/homeware/dining/cat",
          "id": "/homeware/dining/cat",
          "path": "/homeware/dining/cat",
          "url": "/homeware/dining/cat",
        },
      ],
      "header": "Generic Grid",
      "slide1_image": Array [
        Object {
          "asset_file": "1513596567.2856915-Week-44-Department-Homeware-2_24.jpg",
          "canonical_path": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg",
          "created_at": "2017-12-18T11:29:27Z",
          "file_content_content_type": "image/jpeg",
          "file_content_filename": "Week-44-Department-Homeware-2_24.jpg",
          "file_content_size": 354806,
          "id": "157604",
          "image_height": 1483,
          "image_width": 2396,
          "reference": "26418193-2571-4bb8-943b-bb567065c4f8",
          "s3_url": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg",
          "updated_at": "2017-12-18T11:29:27Z",
        },
      ],
      "slide1_text": "shop glassware",
      "slide1_url": Array [
        Object {
          "canonical_path": "/homeware/dining/glassware",
          "id": "/homeware/dining/glassware",
          "path": "/homeware/dining/glassware",
          "url": "/homeware/dining/glassware",
        },
      ],
      "slide2_image": Array [
        Object {
          "asset_file": "1513596567.2856915-Week-44-Department-Homeware-2_25.jpg",
          "canonical_path": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_25.jpg",
          "created_at": "2017-12-18T11:29:27Z",
          "file_content_content_type": "image/jpeg",
          "file_content_filename": "Week-44-Department-Homeware-2_24.jpg",
          "file_content_size": 354806,
          "id": "157604",
          "image_height": 1483,
          "image_width": 2396,
          "reference": "26418193-2571-4bb8-943b-bb567065c4f8",
          "s3_url": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg",
          "updated_at": "2017-12-18T11:29:27Z",
        },
      ],
      "slide2_text": "shop glassware2",
      "slide2_url": Array [
        Object {
          "canonical_path": "/homeware/dining/glassware2",
          "id": "/homeware/dining/glassware",
          "path": "/homeware/dining/glassware",
          "url": "/homeware/dining/glassware",
        },
      ],
      "slide3_image": Array [
        Object {
          "asset_file": "1513596567.2856915-Week-44-Department-Homeware-2_26.jpg",
          "canonical_path": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_26.jpg",
          "created_at": "2017-12-18T11:29:27Z",
          "file_content_content_type": "image/jpeg",
          "file_content_filename": "Week-44-Department-Homeware-2_24.jpg",
          "file_content_size": 354806,
          "id": "157604",
          "image_height": 1483,
          "image_width": 2396,
          "reference": "26418193-2571-4bb8-943b-bb567065c4f8",
          "s3_url": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg",
          "updated_at": "2017-12-18T11:29:27Z",
        },
      ],
      "slide3_text": "shop glassware3",
      "slide3_url": Array [
        Object {
          "canonical_path": "/homeware/dining/glassware3",
          "id": "/homeware/dining/glassware",
          "path": "/homeware/dining/glassware",
          "url": "/homeware/dining/glassware",
        },
      ],
      "slide4_text": "",
    }
  }
>
  <section
    className="o-template-component u-center-align"
  >
    <h1
      className="c-component-header"
    >
      Generic Grid
    </h1>
    <div
      className="o-card-grid o-card-grid--3d-3m"
    >
      <div
        className="o-card-grid__card"
        key="0"
      >
        <Link
          href="/homeware/dining/glassware"
        >
          <a
            href="/homeware/dining/glassware"
          >
            <LazyLoad
              className="u-image-shadow"
              src="https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg"
            >
              <div
                style={
                  Object {
                    "backgroundColor": "#EEEFF3",
                    "height": 0,
                    "paddingBottom": "NaN%",
                    "width": "100%",
                  }
                }
              >
                <FadeIn
                  offset={150}
                  once={true}
                  render={[Function]}
                >
                  <LazyLoad
                    offset={150}
                    once={true}
                    overflow={false}
                    resize={false}
                    scroll={true}
                    unmountIfInvisible={false}
                  >
                    <div
                      className="lazyload-placeholder"
                      style={
                        Object {
                          "height": undefined,
                        }
                      }
                    />
                  </LazyLoad>
                </FadeIn>
              </div>
            </LazyLoad>
            <p
              className="o-card-grid__title"
            >
              shop glassware
            </p>
          </a>
        </Link>
      </div>
      <div
        className="o-card-grid__card"
        key="1"
      >
        <Link
          href="/homeware/dining/glassware2"
        >
          <a
            href="/homeware/dining/glassware2"
          >
            <LazyLoad
              className="u-image-shadow"
              src="https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_25.jpg"
            >
              <div
                style={
                  Object {
                    "backgroundColor": "#EEEFF3",
                    "height": 0,
                    "paddingBottom": "NaN%",
                    "width": "100%",
                  }
                }
              >
                <FadeIn
                  offset={150}
                  once={true}
                  render={[Function]}
                >
                  <LazyLoad
                    offset={150}
                    once={true}
                    overflow={false}
                    resize={false}
                    scroll={true}
                    unmountIfInvisible={false}
                  >
                    <div
                      className="lazyload-placeholder"
                      style={
                        Object {
                          "height": undefined,
                        }
                      }
                    />
                  </LazyLoad>
                </FadeIn>
              </div>
            </LazyLoad>
            <p
              className="o-card-grid__title"
            >
              shop glassware2
            </p>
          </a>
        </Link>
      </div>
      <div
        className="o-card-grid__card"
        key="2"
      >
        <Link
          href="/homeware/dining/glassware3"
        >
          <a
            href="/homeware/dining/glassware3"
          >
            <LazyLoad
              className="u-image-shadow"
              src="https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_26.jpg"
            >
              <div
                style={
                  Object {
                    "backgroundColor": "#EEEFF3",
                    "height": 0,
                    "paddingBottom": "NaN%",
                    "width": "100%",
                  }
                }
              >
                <FadeIn
                  offset={150}
                  once={true}
                  render={[Function]}
                >
                  <LazyLoad
                    offset={150}
                    once={true}
                    overflow={false}
                    resize={false}
                    scroll={true}
                    unmountIfInvisible={false}
                  >
                    <div
                      className="lazyload-placeholder"
                      style={
                        Object {
                          "height": undefined,
                        }
                      }
                    />
                  </LazyLoad>
                </FadeIn>
              </div>
            </LazyLoad>
            <p
              className="o-card-grid__title"
            >
              shop glassware3
            </p>
          </a>
        </Link>
      </div>
    </div>
    <Link
      className="o-template-component__cat-button o-button o-button--primary"
      href="/homeware/dining/cat"
    >
      <a
        className="o-template-component__cat-button o-button o-button--primary"
        href="/homeware/dining/cat"
      >
        Cat Button
      </a>
    </Link>
  </section>
</GenericGrid>
`;
